### ISSUE TITLE
memsync - TypeError

### ISSUE THREAD

**Issue by jtlz2** [pleiszenburg/zugbruecke#29]
I am just learning how to use `memsync` (on your advice, for which thanks :) ) - for a pointer to an array nested within a structure defined by me. Following the documentation, for a structure that's the third argument (i.e. index 2), I've tried:

```
    func.memsync = [
        {
                'p': [2,'a'],
                'l': [2,'length_of_a'],
                '_t': ctypes.c_ubyte
        }
        ]
```

This gives:

  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/zugbruecke-0.0.8-py3.6.egg/zugbruecke/core/routine_client.py", line 114, in __call__
    self.__configure__()
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/zugbruecke-0.0.8-py3.6.egg/zugbruecke/core/routine_client.py", line 171, in __configure__
    self.memsync_handle = self.apply_memsync_to_argtypes_definition(self.__memsync__, self.argtypes_d)
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/zugbruecke-0.0.8-py3.6.egg/zugbruecke/core/arg_definition.py", line 69, in apply_memsync_to_argtypes_definition
    arg_type = arg_type['_fields_'][path_element]
TypeError: list indices must be integers or slices, not str

I think the loop index path_element might need to be over `range(len(segment['p'][1:]))` (twice), but then I get an error in memory.py.

Am I doing something stupid? Perhaps a clean install would help me?

Thanks!

**Comment by s-m-e** [comment]
No, you're not doing something stupid. This appears to be a legitimate bug. I'll look into it.

**Comment by s-m-e** [comment]
Fixed in 19445c5 (development branch).

@jtlz2 You can install / try it like this: `pip install git+https://github.com/pleiszenburg/zugbruecke.git@develop`
